lib/ecto_job/job_queue.ex

```diff
@@ -307,10 +308,43 @@ defmodule EctoJob.JobQueue do
   end
 
   @doc """
-  Computes the expiry time for an `"IN_PROGRESS"` job based on the current time and attempt counter.
+  Transitions a job from `"IN_PROGRESS"` to `"RETRYING"`.
+
+  Updates the state to `"RETRYING"` and changes the schedule time to
+  differentiate an expired job from one that had an exception or an error.
+
+  """
+  @spec update_job_to_retrying(repo, job, DateTime.t(), integer) ::
+          {:ok, Ecto.Schema.t()} | {:error, String.t}
+  def update_job_to_retrying(repo, job  = %schema{}, now, timeout_ms) do
+    {count, results} =
+      repo.update_all(
+        Query.from(
+          j in schema,
+          where: j.id == ^job.id,
+          where: j.state == "IN_PROGRESS"
+        ),
+        [
+          set: [
+            state: "RETRYING",
+            schedule: progress_time(now, job.attempt + 1, timeout_ms),
+            updated_at: now
+          ]
+        ],
+        returning: true
```

Comment:
`returning: true` is going to be deprecated in Ecto 3.0.

Reply:
:+1: I'll change it for `select: j` then